dataset_interfaces/interface.py

```diff
@@ -265,6 +265,12 @@ class DatasetInterface(ABC):
     def __post_init__(self):
         self.random = Random(self.seed)
 
+    def __getattribute__(self, item):
```

Comment:
In what situation is this needed? `__post_init__` should take care of the setting of the seed, no?

Reply:
I guess it depends on the behaviour we want. I'd expect `generate_examples` to be called only once per dataset, or at least be expected to generate the same samples if it's called twice.

Reply:
Hmmm, I see your point. I guess we should expect it to generate the same tests each call to `generate_examples`. In this case I would remove the mention of the random from `__post_init__` if that won't cause any problems.